### ISSUE TITLE
chrome defunct processes

### ISSUE THREAD

**Issue by TomVanBerlo** [boyney123/garie-lighthouse#5]
I recently installed garie via the docker-compose file included in https://github.com/boyney123/garie.
But after a while it stops collecting metrics.
I pinned down the issue to zombie processes from 'chrome' and 'cat' using all resources.
After some more searching it turns out that the 'npm' process runs as pid 1. This pid is normally reserved for the init process that reaps zombies.
A better explanation can be found at https://github.com/Yelp/dumb-init.
My proposed solution is to install dumb-init in the Dockerfile.
```
diff --git a/Dockerfile b/Dockerfile
index e649d43..8e9b599 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -12,7 +12,7 @@ RUN \
     wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
     sh -c 'echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' && \
     apt-get update && \
-    apt-get install -y google-chrome-stable
+    apt-get install -y google-chrome-stable dumb-init

 COPY . .
```

**Comment by GerardSmit** [comment]
Is this relevent to the errors in my logging, or should I create a new issue for this?

```
error: Failed to get data for https://example.com {"stack":"Error: spawnSync /bin/sh ENOSPC
    at _errnoException (util.js:1022:11)
    at spawnSync (child_process.js:579:20)
    at execSync (child_process.js:635:13)
    at findChromeExecutables (/usr/src/garie-lighthouse/node_modules/chrome-launcher/dist/chrome-finder.js:201:25)
    at desktopInstallationFolders.forEach.folder (/usr/src/garie-lighthouse/node_modules/chrome-launcher/dist/chrome-finder.js:87:46)
    at Array.forEach (<anonymous>)
    at Object.linux (/usr/src/garie-lighthouse/node_modules/chrome-launcher/dist/chrome-finder.js:86:32)
    at Launcher.<anonymous> (/usr/src/garie-lighthouse/node_modules/chrome-launcher/dist/chrome-launcher.js:139:80)
    at Generator.next (<anonymous>)
    at /usr/src/garie-lighthouse/node_modules/chrome-launcher/dist/chrome-launcher.js:12:71","timestamp":"2018-11-25T14:30:00.192Z"}
```

Thanks in advance.

**Comment by GerardSmit** [comment]
I've build your fork and can confirm that the errors are relevent to this issue.

@boyney123 I don't know if you have time to maintain this project, but this is quite an important issue since it spawns many zombie processes that I even couldn't access my server.

**Comment by gschoenberger** [comment]
Any update on this issue? We are hitting the same issue...

**Comment by boyney123** [comment]
Hey @gschoenberger @GerardSmit @TomVanBerlo 

Sorry its taken so long, I have merged in the PR to fix this and released the Docker image 2.0 👍

**Comment by boyney123** [comment]
Let me know if you still have issues.

On another note if anybody wants to help maintain Garie, feel free to get in contact! :)
